tests/ssl-opt.sh

```diff
@@ -3967,20 +3965,8 @@ run_test    "Authentication: client no cert, ssl3" \
 MAX_IM_CA='8'
 MAX_IM_CA_CONFIG="$( get_config_value_or_default MBEDTLS_X509_MAX_INTERMEDIATE_CA )"
 
-if [ -n "$MAX_IM_CA_CONFIG" ] && [ "$MAX_IM_CA_CONFIG" -ne "$MAX_IM_CA" ]; then
-    printf "The configuration file contains a value for the configuration of\n"
-    printf "MBEDTLS_X509_MAX_INTERMEDIATE_CA that is different from the script’s\n"
-    printf "test value of ${MAX_IM_CA}. \n"
-    printf "\n"
-    printf "The tests assume this value and if it changes, the tests in this\n"
-    printf "script should also be adjusted.\n"
-    printf "\n"
-
-    exit 1
-fi
-
 requires_full_size_output_buffer
-requires_config_value_at_least "MBEDTLS_SSL_MAX_CONTENT_LEN" 4096
+requires_config_value_at_least "MBEDTLS_X509_MAX_INTERMEDIATE_CA" 8
```

Comment:
It should be _precisely_ 8 here. Could you either combine `requires_config_value_at_least/most` or add a new helper function `requires_config_value` that asks for an exact value?